app/src/main/java/io/neurolab/activities/DataLoggerActivity.java

```diff
@@ -61,8 +69,53 @@ private void showLoggedDataList(File appDir) {
             dataloggerRecyclerView.setLayoutManager(linearLayoutManager);
             dataloggerRecyclerView.setAdapter(adapter);
         } else {
-            dataloggerRecyclerView.setVisibility(View.GONE);
-            noLoggedView.setVisibility(View.VISIBLE);
+
+            if (!appDir.exists() && !appDir.isDirectory()) {
+                appDir.mkdirs();
+            }
+
+            noLoggedView.setVisibility(View.GONE);
+
+            readWriteData("sample1", appDir);
+            readWriteData("sample2", appDir);
+            readWriteData("sample3", appDir);
+            readWriteData("sample4", appDir);
+
+            DataLoggerListAdapter adapter = new DataLoggerListAdapter(this, filesList, flag);
+            LinearLayoutManager linearLayoutManager = new LinearLayoutManager(
+                    this, LinearLayoutManager.VERTICAL, false);
+            dataloggerRecyclerView.setLayoutManager(linearLayoutManager);
+            dataloggerRecyclerView.setAdapter(adapter);
+        }
+    }
+
+    private void readWriteData(String filename, File appDir) {
```

Comment:
@dipu989 Check the `FilePathUtils` class, it has similar utility functions for file creation and stuff. Please use from there, reduces redundancy. Otherwise changes intended look good :+1:

Reply:
@jddeep  The Pathsetup function works well from FilePathUtils. Also File's getting created but is unable to read anything.
As of now I have code like this : -
```
 String uriPath = "android.resource://" + getPackageName() + "/raw/sample1";
            Uri uri = Uri.parse(uriPath);
            String str = getRealPath(getApplicationContext(), uri);
            saveData(str);
```
There is a bit doubt over RecordData(String data), as it demands data as parameter to write it to another CSV file and I have csv files as parameter.
Any suggestions over where I am getting wrong in reading data from the location?

Reply:
@dipu989 Use the `setupPath` function from the `FilePathUtil` class. Then move your `readwriteData` function to that class and use it from there. Utility functions should be kept in utility classes.